From 13716da489f4faf13edca5c45614904a9cb5e1a8 Mon Sep 17 00:00:00 2001
From: Martin Kibera <martin.kibera.n@gmail.com>
Date: Sun, 26 Oct 2025 11:56:48 +0300
Subject: [PATCH 05/31] replace b720p with just 720p

---
 pug/settings.pug | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pug/settings.pug b/pug/settings.pug
index d56de25..30f1157 100644
--- a/pug/settings.pug
+++ b/pug/settings.pug
@@ -71,8 +71,7 @@ block content
           label: "Preferred qualities",
           description: "All qualities are available on the watch page. This defines their sort order.",
           options: [
-            {value: "1", text: "b720p"},
-            {value: "0", text: "720p"},
+            {value: "1", text: "720p"},
             {value: "4", text: "360p"},
             {value: "5", text: "Best possible"},
             {value: "2", text: "Best <=1080p"},
-- 
2.52.0

