From 5ed1ac5748c54c1006c7fdb1aa158d8ed4f70b5c Mon Sep 17 00:00:00 2001
From: Martin Kibera <martin.kibera.n@gmail.com>
Date: Mon, 17 Nov 2025 19:59:17 +0300
Subject: [PATCH] fix logging spam

---
 html/static/js/chapter-highlight.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/html/static/js/chapter-highlight.js b/html/static/js/chapter-highlight.js
index d87ecd3..194cf81 100644
--- a/html/static/js/chapter-highlight.js
+++ b/html/static/js/chapter-highlight.js
@@ -24,6 +24,7 @@ function getCurrentChapter(time) {
 }
 
 function updateHighlight() {
+	if (!video) return
 	const currentChapter = getCurrentChapter(video.currentTime)
 	if (currentChapter !== lastChapter) {
 		if (lastChapter) lastChapter.link.removeClass("timestamp--active")
-- 
2.53.0

